### PR TITLE
Update to latest swift, uiAlertViewControllers #6

### PR DESCRIPTION
**Pull request by MacMeDan**
I replaced the Alert View with an AlertViewController. Detect wither or not the app is running on a simulator or a real device and prompt the user to enter the servers IP address if they are on a real device. Added a button so they could get the prompt again. Updated to the latest swift syntax.

### CODE REVIEW

File: TicTacIOiOS/ViewController.swift [hunk 1]
```diff
@@ -18,32 +18,48 @@ class ViewController: UIViewController, UIAlertViewDelegate {
...
     @IBOutlet weak var btn8:UIButton!
     @IBOutlet weak var label:UILabel!
-    let socket = SocketIOClient(socketURL: NSURL(string:"http://localhost:8900")!)
+    
     var name: String?
     var resetAck: SocketAckEmitter?
     let backgroundGrad = CAGradientLayer()
     
+    var inputTextField: UITextField?
+    var socket: SocketIOClient!
```
Comment (nuclearace) [inline TicTacIOiOS/ViewController.swift]: This could be optional, and then use chaining.

Comment (MacMeDan) [inline TicTacIOiOS/ViewController.swift]: So I looked into this. If I do it as an optional on every call to the socket I have to force unwrap it. Which I feel is more un safe. I may be missing something on how optional chaining would be an advantage. I would love to understand more.

File: TicTacIOiOS/ViewController.swift [hunk 2]
```diff
@@ -119,8 +143,9 @@ class ViewController: UIViewController, UIAlertViewDelegate {
         default:
             coord = (-1, -1)
         }
-        
-        socket.emit("playerMove", coord.x, coord.y)
+        if let socket = socket {
+            socket.emit("playerMove", coord.x, coord.y)
```
Comment (nuclearace) [inline TicTacIOiOS/ViewController.swift]: Like here instead of an if let you could just do `socket?.emit("playerMove", coord.x, coord.y)`

**Comment by MacMeDan** [comment]
Is there anything else I need to do here.

**Pull request merged**